src/main/kotlin/uk/gov/dluhc/printapi/database/repository/PrintDetailsRepository.kt

```diff
@@ -42,8 +42,9 @@ class PrintDetailsRepository(client: DynamoDbEnhancedClient, tableConfig: Dynamo
     }
 
     fun getAllByStatus(status: Status): List<PrintDetails> {
-        // TODO: This is temporary as dynamodb has given me enough headaches for now
-        return table.scan().flatMap { it.items() }.filter { it.status == Status.PENDING_ASSIGNMENT_TO_BATCH }
+        val index = table.index(STATUS_BATCH_ID_INDEX_NAME)
+        val queryConditional = QueryConditional.keyEqualTo(key(status.toString()))
+        return index.query(queryConditional).flatMap { it.items() }.toList()
```

Comment:
If a default batch Id is not set, use:

`        val expression = Expression.builder()
            .expression("#print_status = :status")
            .expressionNames(mapOf(Pair("#print_status","status")))
            .expressionValues(mapOf(
                Pair(":status", AttributeValue.builder().s(Status.PENDING_ASSIGNMENT_TO_BATCH.toString()).build())))
            .build()
        val scanFiltered = table.scan(ScanEnhancedRequest.builder().filterExpression(expression).build()).items().toList()`

Reply:
If an unassigned batch id can be set and thus an index query can be done using:
```
        val index = table.index(STATUS_BATCH_ID_INDEX_NAME)
        val queryConditional = QueryConditional.keyEqualTo(key(status.toString()))
        return index.query(queryConditional).flatMap { it.items() }.toList()
```